src/main/java/cn/nukkit/form/window/FormWindow.java

```diff
@@ -1,13 +1,32 @@
 package cn.nukkit.form.window;
 
+import cn.nukkit.form.handler.FormResponseHandler;
 import cn.nukkit.form.response.FormResponse;
+import com.google.gson.ExclusionStrategy;
+import com.google.gson.FieldAttributes;
 import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import it.unimi.dsi.fastutil.objects.ObjectArrayList;
+
+import java.lang.reflect.Modifier;
+import java.util.List;
 
 public abstract class FormWindow {
 
-    private static final Gson GSON = new Gson();
-    
+    private static final Gson GSON = new GsonBuilder().addSerializationExclusionStrategy(new ExclusionStrategy() {
+        @Override
+        public boolean shouldSkipField(FieldAttributes f) {
+            return f.getName().equals("handlers");
+        }
+
+        @Override
+        public boolean shouldSkipClass(Class<?> clazz) {
+            return false;
+        }
+    }).create();
+
     protected boolean closed = false;
+    protected final List<FormResponseHandler> handlers = new ObjectArrayList<>();
```

Comment:
Did you try adding the `transient` keyword to this variable? It should make GSON ignore the value so you don't need to do that hack above.

Reply:
Hmm stackoverflow says no